### PR TITLE
doc/rados/operations/add-or-rm-mons: simplify the steps to add a mon

### PR DESCRIPTION
**Pull request by tchaikov**
actually "ceph mon add" is not necessary, so remove it.

http://tracker.ceph.com/issues/12620 Fixes: #12620
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

File: doc/rados/operations/add-or-rm-mons.rst [hunk 1]
```diff
@@ -126,6 +121,14 @@ on ``mon.a``).
 
 	ceph-mon -i {mon-id} --public-addr {ip:port}
 
+#. Add the new monitor to the list of monitors for you cluster (runtime). This enables
+   other nodes to use this monitor during their initial startup. Please
+   note, following command will wait until a quorum is formed. So, if
```
Comment (dachary) [inline doc/rados/operations/add-or-rm-mons.rst]: the following

**Comment by tchaikov** [comment]
thanks @dachary updated and re-pushed.

**Comment by trhoden** [comment]
@tchaikov So, does this mean that if you start the new monitor before doing `mon add`, you shouldn't run into problems when going from 1 mon to 2?

I'm wondering if I should switch the steps in ceph-deploy here:

https://github.com/ceph/ceph-deploy/blob/master/ceph_deploy/hosts/common.py#L149-L180

**Comment by tchaikov** [comment]
> So, does this mean that if you start the new monitor before doing `mon add`, you shouldn't run into problems when going from 1 mon to 2?

@trhoden yeah. see also http://tracker.ceph.com/issues/12620

File: doc/rados/operations/add-or-rm-mons.rst [hunk 2]
```diff
@@ -126,6 +121,14 @@ on ``mon.a``).
 
 	ceph-mon -i {mon-id} --public-addr {ip:port}
 
+#. Add the new monitor to the list of monitors for you cluster (runtime). This enables
```
Comment (trhoden) [inline doc/rados/operations/add-or-rm-mons.rst]: s/you/your   (your cluster)

sorry to nitpick.  That was an old/existing typo.  Might as well fix it now!  :)

**Comment by trhoden** [comment]
Already tested the change in ceph-deploy, and it does indeed solve the problems when going from 1 to 2 monitors.

With the one nitpick to fix:

`Reviewed-By: Travis Rhoden <trhoden@redhat.com>`

**Comment by liewegas** [comment]
Note that the easiest way to do this is one step:

 ceph-deploy mon create NEWMON

Assuming the key is in place (should be in ~), the new mon will negotiate 
with the old one(s) to add itself.. no other 'ceph mon add ...' commands 
are necessary.

**Comment by tchaikov** [comment]
@liewegas, as 'ceph-deploy mon create NEWMON' will basically just 'ceph -c /etc/ceph/ceph.conf start mon.{hostname}'. i think, we can simply remove this step to avoid unnecessary confusions. what do you think?

**Comment by liewegas** [comment]
yeah... but test it first, i'm not certain i'm remember correctly :)

File: doc/rados/operations/add-or-rm-mons.rst (hunk 2 above)
Comment (tchaikov) [inline doc/rados/operations/add-or-rm-mons.rst]: fixed =)

**Comment by tchaikov** [comment]
@liewegas sure. tested, and we (will) have a test https://github.com/tchaikov/ceph/blob/b48dfd763f4c0e4fb70c437ca4020ec383ece56e/src/test/mon/misc.sh#L88-L121 for this.

**Comment by tchaikov** [comment]
@jecluis what do you think?

**Comment by jecluis** [comment]
For clarity's sake, I think mentioning that 'ceph mon add' is not needed and why would make this patch just perfect :) Other than that,

`Reviewed-by: Joao Eduardo Luis <joao@suse.de>`

**Comment by tchaikov** [comment]
thanks @jecluis, i will amend the commit log to elaborate on the what and why.

**Comment by tchaikov** [comment]
@jecluis @dachary does this change look good to you?

**Comment by jecluis** [comment]
@tchaikov maybe I wasn't clear on my last comment, but I meant mentioning this on the docs themselves. We're removing something users should be quite accustomed to (`ceph mon add`) and not explaining why. I believe mentioning that it is not needed, and why, should go on the docs themselves. Just a brief explanation on the sorts of

> The monitor will automatically negotiate with existing monitors and join the quorum, without any further intervention needed.

**Comment by tchaikov** [comment]
@jecluis sorry, i still don't quite get what you mean, so you suggest we keep the removed step, and add a note, like "this step is not needed. " and plus the brief explanation you put in the comment?

that would introduce some confusion, imho. if i were are a brave and curious user, i could be tempted to try out the steps (including the not needed steps) to add a new monitor on a single-monitor cluster, and inevitably, would run into the "expected-behaviour".

probably, it would be better to move this "ceph mon add" out of the order list of "Adding a Monitor (Manual)" section, and start a new paragraph to put these explanations for users who are expecting the "ceph mon add" command?

**Comment by jecluis** [comment]
@tchaikov please disregard everything on my last comment. I must be high on something, as I completely missed that the next step in the list quite plainly states `#. Start the new monitor and it will automatically join the cluster.`

So yeah, no wonder my comment was confusing :)

**Pull request merged**